ykcompile/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+        // up registers again, and allocate additional locals on the stack. Though, ultimately we
+        // probably want to implement a proper register allocator, e.g. graph colouring or linear
+        // scan.
+        match self.assigned_regs.get(&l) {
+            Some(u) => {
+                *u
+            },
+            None => {
+                if l == 0 {
+                    // Local 0 always maps to the return register RAX
+                    0
+                }
+                else {
+                    let reg = self.available_regs
+                        .pop()
+                        .expect("Can't allocate more than 7 locals yet!");
+                    self.assigned_regs.insert(l, reg);
+                    reg
+                }
+            }
+        }
+    }
+
+    pub fn local_usize(&mut self, a: u32, c: usize) {
+        let reg = self.local_to_reg(a);
+        dynasm!(self.asm
+            ; mov Rq(reg), c as i32
+        );
+    }
+
+    pub fn local_u8(&mut self, a: u32, c: u8) {
+        let reg = self.local_to_reg(a);
+        dynasm!(self.asm
+            ; mov Rq(reg), c as i32
+        );
+    }
+
+    fn local_local(&mut self, var1: u32, var2: u32) {
+        let lreg = self.local_to_reg(var1);
+        let rreg = self.local_to_reg(var2);
+        dynasm!(self.asm
+            ; mov Rq(lreg), Rq(rreg)
+        );
+    }
+
+    fn statement(&mut self, stmt: &Statement) {
+        match stmt {
+            Statement::Assign(l, r) => {
+                let local = l.local.0;
+                match r {
+                    Rvalue::Use(Operand::Place(p)) => {
+                        self.local_local(local, p.local.0)
+                    },
+                    Rvalue::Use(Operand::Constant(c)) => {
+                        match c {
+                            Constant::Int(ci) => {
+                                match ci {
+                                    ConstantInt::UnsignedInt(UnsignedInt::U8(i)) => {
+                                        self.local_u8(local, *i)
+                                    },
+                                    ConstantInt::UnsignedInt(UnsignedInt::Usize(i)) => {
+                                        self.local_usize(local, *i)
+                                    },
+                                    e => panic!("SignedInt, etc: {}", e)
+                                }
+                            }
+                            Constant::Bool(b) => {
+                                self.local_u8(local, *b as u8)
+                            }
+                            c => panic!("Not implemented: {}", c)
+                        }
+                    }
+                    unimpl => panic!("Not implemented: {:?}", unimpl)
+                };
+            },
+            Statement::Return => {},
+            Statement::Nop => {},
+            Statement::Unimplemented(mir_stmt) => println!("Can't compile: {}", mir_stmt)
```

Comment:
This should surely cause a `panic`, not try and carry on after a `println`?

Reply:
This was already fixed.